### PR TITLE
CRUD for Resources

### PR DESCRIPTION
**Pull request by aaronabraham311**
## Overview
Creating ambulances and hospitals resource tables and designing functionality for querying, adding, deleting and updating tables. 

[Notion task](https://www.notion.so/uwblueprintexecs/hospitals-ambulance-db-setup-e7773a02f2d94a1a8b432dfdac7a4db8)

## Changes
* Created migrations and seeders for the hospital table: As of now, the only field defined for the hospital table is the name of the hospital. This also involved:
    * Schema updated in `schema/hospital.js`
    * Resolvers updated in `resolvers/hospital.js`
    * Set up models in `models/hospital.js`
    * Schema updated in `schema/ambulance.js`
    * Resolvers updated in `resolvers/ambulance.js`
    * Set up models in `models/ambulance.js`
* Linked changes in `graphql.js`
* Created update and deletion resolvers in `schema/hospital.js`, `resolvers/hospital.js`, `schema/ambulance.js`, and `resolvers/ambulance.js`
    * Note: The return type of resolvers is an object of the dataset itself. Would love to try to find a way to not return an object but rather an integer or nothing at all.
* Error handling for updates (suggesting that we make some error file to pull in common error outputs)
* Changed schema to reflect inbuilt Sequelize function error handling:
    * db.delete: Returns 1 if record deleted, 0 if deleted
    * Update resolvers: Returning Hospital/Ambulance object if object has been successfully updates; otherwise we will throw a custom error

## Screenshots
General screenshot
image

Error handling for updating on non-existing hospitals
Screen Shot 2020-03-09 at 8 49 59 PM

Output for deletion of existing hospitals
Screen Shot 2020-03-09 at 8 51 52 PM

Output for deletion of non-existing hospitals
Screen Shot 2020-03-09 at 8 53 16 PM

## Testing
* Manual testing via GraphIQL

## Checklist
- [x] Change title to be more descriptive (shouldn't be just branch name)
- [x] Update `Notion task` link
- [x] `Overview`
- [x ] \(Optional) Add inline comments in diff
- [x] \(Optional) `Screenshots`
- [x] \(Optional) `Testing` (N/A if nothing changed)
- [x] Request at least 2 reviewers
- [x] Assign yourself to the PR
- [x ] Make sure build passes

@uwblueprint/paramedics

### CODE REVIEW

**Comment by JasonYG** [comment]
Creates the `ambulances` and `hospitals` tables, as well as adds CRUD mutations.  I tested querying both tables, as well as creating, updating, and deleting rows.

File: graphql.js [hunk 1]
```diff
@@ -38,8 +42,8 @@ const resolvers = {
 };
 
 const schema = makeExecutableSchema({
-    typeDefs: [ scalars, Query, Mutation, userSchema, eventSchema ],
-    resolvers: merge(resolvers, userResolvers, eventResolvers),
+  typeDefs: [ scalars, Query, Mutation, userSchema, eventSchema, hospitalSchema, ambulanceSchema],
```
Comment (danielchen115) [inline graphql.js]: (SUPER nitpicky i know but) can we add the space back after `ambulanceSchema` :)

**Comment by danielchen115** [comment]
Looks great, tested all the queries. Good work!  One quick question: What are the changes to package-lock.json for? It doesn't look like there are any new libraries being installed.

File: migrations/20200213003503-create-hospitals.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('hospitals', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
+        type: Sequelize.STRING
```
Comment (petorsong) [inline migrations/20200213003503-create-hospitals.js]: constraining these to be non-null would be a good idea (cc: @danielchen115)

File: migrations/20200213013205-create-ambulances.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('ambulances', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      vehicleNumber: {
+        type: Sequelize.INTEGER
```
Comment (petorsong) [inline migrations/20200213013205-create-ambulances.js]: ditto for non-null

File: resolvers/ambulance.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  Query: {
+    ambulances: () => db.ambulance.findAll(),
+    ambulance: (obj, args, context, info) => db.ambulance.findByPk(args.id)
+  },
+  Mutation: {
+    addAmbulance: (parent, args) => {
+        const ambulance = db.ambulance.create({
+            vehicleNumber: args.vehicleNumber
+        });
+        return ambulance;
```
Comment (petorsong) [inline resolvers/ambulance.js]: nitpick/code style: could be simplified
```suggestion
        return db.ambulance.create({
            vehicleNumber: args.vehicleNumber
        });
```

File: graphql.js (hunk 1 above)
Comment (aaronabraham311) [reply graphql.js]: Done!

**Comment by aaronabraham311** [comment]
> Looks great, tested all the queries. Good work! One quick question: What are the changes to package-lock.json for? It doesn't look like there are any new libraries being installed.

 @danielchen115 I had to download a mySQL package when I was testing on prod. I believe the changes are from that. Do I need to reverse them?

**Comment by danielchen115** [comment]
> > Looks great, tested all the queries. Good work! One quick question: What are the changes to package-lock.json for? It doesn't look like there are any new libraries being installed.
> 
> @danielchen115 I had to download a mySQL package when I was testing on prod. I believe the changes are from that. Do I need to reverse them?

In that case, I think it's best we remove the changes for that. We no longer need mysql2 for prod.
run `git checkout master -- package-lock.json`

**Comment by aaronabraham311** [comment]
> > > Looks great, tested all the queries. Good work! One quick question: What are the changes to package-lock.json for? It doesn't look like there are any new libraries being installed.
> > 
> > 
> > @danielchen115 I had to download a mySQL package when I was testing on prod. I believe the changes are from that. Do I need to reverse them?
> 
> In that case, I think it's best we remove the changes for that. We no longer need mysql2 for prod.
> run `git checkout master -- package-lock.json`

@danielchen115 I have addressed this in my latest commit

**Pull request merged**